src/components/InputStepCounter/InputStepCounter.tsx

```diff
@@ -144,10 +144,8 @@ const StepCounter = ({
     }
   }, [localValue, useLocalValue, value])
 
-  const baseCurrencyPriceInUSDC = useUSDCPrice(currencyB ?? undefined)
-  const baseCurrencyPriceInUSDCNumber = Number(baseCurrencyPriceInUSDC?.toFixed(4))
-  const baseCurrencyPriceIsNotNaN = !isNaN(baseCurrencyPriceInUSDCNumber)
   const quoteCurrencyAmount = Number(value)
+  const baseCurrencyUSDPrice = useCurrencyUSDPrice(currencyB ?? undefined, quoteCurrencyAmount)
```

Comment:
ooh hmm a few typing issues are making this difficult to use the `CurrencyAmount` format: `currencyB` might be undefined, and `quoteCurrencyAmount` is a decimal instead of a `BigInt` as the constructor expects. Maybe cleaner to do it without `CurrencyAmount`?

Reply:
oooh nvm, i found a helper function that gives me what i need: `tryParsedAmount` which converts the input string into `CurrencyAmount`!